cheroot/test/test_wsgi.py

```diff
@@ -0,0 +1,59 @@
+"""Test wsgi."""
+
+import threading
+from concurrent.futures.thread import ThreadPoolExecutor
+
+import pytest
+import portend
+import requests
+from requests_toolbelt.sessions import BaseUrlSession as Session
+from jaraco.context import ExceptionTrap
+
+from cheroot import wsgi
+
+
+@pytest.fixture
+def simple_wsgi_server():
+    """Fucking simple wsgi server fixture (duh)."""
+    port = portend.find_available_local_port()
+
+    def app(environ, start_response):
+        status = '200 OK'
+        response_headers = [('Content-type', 'text/plain')]
+        start_response(status, response_headers)
+        return [b'Hello world!']
+
+    host = '::'
+    addr = host, port
+    server = wsgi.Server(addr, app)
```

Comment:
```suggestion
    server = wsgi.Server(addr, app, timeout=0, accepted_queue_timeout=0)
```

@the-allanc like this?

Reply:
`ConnectionManager.expire` looks at `server.timeout` to work out whether the socket should be closed - so I think that's the only setting that we should try changing (and limit the effect to just this one test).

I'd probably be more inclined to increase the default timeout to a high number, rather than zero (just because I'm not sure what that will do).

Reply:
Interesting, changing this revealed some bug: https://github.com/cherrypy/cheroot/pull/277/checks?check_run_id=576581037#step:14:413. It now returns HTTP 500.

Reply:
And it's now 465 failures, not 10 like before: https://travis-ci.com/github/cherrypy/cheroot/jobs/318105925#L441

Reply:
> Interesting, changing this revealed some bug: [#277 (checks)](https://github.com/cherrypy/cheroot/pull/277/checks?check_run_id=576581037#step:14:413). It now returns HTTP 500.

Oh, so docs for `io.BufferedReader.read()` don't mention the possibility of returning `None` but https://bugs.python.org/issue35869 suggests that this may happen when in non-blocking mode (hence `timeout=0`)

Reply:
Ideally, it shouldn't be necessary to update this test as it passes on `cheroot < 8.1` (introduction of request-based workers). If the introduction of request-based workers requires the users to supply a non-default parameter where it was not required before, then we'll want to make that explicit in the changelog and give users (and downstream consumers like CherryPy) guidance on how best adapt to this new behavior and we should strongly consider making this parameter the default and adding cautions about failure modes when the current default values are used.

Remember, this test is the confirmed repro reported in the issue. If this test is changed, we'll have to revisit the issue and see what implications this altered test has on that evaluation.

Reply:
_me waves at Jason_
> Ideally, it shouldn't be necessary to update this test as it passes on cheroot < 8.1

To clarify - the test passes for me locally, but under some setups, it seems to fail (mostly on MacOS machines, according to that test run). Are we sure that the test you added will work on the same machines using cheroot 8.1?

My expectation would be that the tests would be likely to fail in the same way - my theory is that the connections / sockets are being closed anyway, as it then exceeds the socket timeout.

Reply:
I'm not sure the test works reliably. I only tested it a few times, probably Python 3.8 and 2.7 on macOS.

Reply:
I'll try cherry-picking that test onto 8.0.x so we can establish a baseline for the test and validate it against the suite of environments in CI.

Reply:
@jaraco I'm not sure you ran exactly the same test: this one uses `concurrent.futures` which is not a part of Python 2. Did you have some other version of it?

Reply:
I did not, and I suspect I did not run the test against Python 2. But now we have [evidence that the tests pass on Python 2.7-3.6](https://github.com/cherrypy/cheroot/runs/579811289).